frontend/src/components/pages/UserManagement/index.tsx

```diff
@@ -15,68 +19,202 @@ import {
 import React from "react";
 
 import DonorAPIClient from "../../../APIClients/DonorAPIClient";
+import UserAPIClient from "../../../APIClients/UserAPIClient";
+import VolunteerAPIClient from "../../../APIClients/VolunteerAPIClient";
+import { Role, Status } from "../../../types/AuthTypes";
 import { DonorResponse } from "../../../types/DonorTypes";
+import { VolunteerResponse } from "../../../types/VolunteerTypes";
+import { UserMgmtTableRecord } from "./types";
 
 const UserManagementPage = (): JSX.Element => {
   const [donors, setDonors] = React.useState([] as DonorResponse[]);
+  const [volunteers, setVolunteers] = React.useState([] as VolunteerResponse[]);
+  const [users, setUsers] = React.useState([] as UserMgmtTableRecord[]); // both donors and volunteers
   const [search, setSearch] = React.useState("");
+  const [selectedFilter, setSelectedFilter] = React.useState("all");
+  const [dataChanged, setDataChanged] = React.useState<0 | 1>(0);
 
-  const tableData = React.useMemo(() => {
-    if (!search) return donors;
-    return donors.filter(
-      (donor) =>
-        donor.businessName.toLowerCase().includes(search.toLowerCase()) ||
-        donor.email.toLowerCase().includes(search.toLowerCase()) ||
-        donor.firstName
-          .toLowerCase()
-          .concat(" ", donor.lastName.toLowerCase())
-          .includes(search.toLowerCase()) ||
-        donor.phoneNumber.toLowerCase().includes(search.toLowerCase()),
-    );
-  }, [search, donors]);
+  const filterOptions = [
+    {
+      value: "all",
```

Comment:
small nit but I think we reference these values as just plain string a few times for us to maybe extract it into a enum? 
```suggestion
      value: "all",
```